packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -0,0 +1,265 @@
+      $Shape<MetadataBackupJob>,
+      ?boolean
+    ) => Promise<MetadataBackupJob>,
+    removeJob: (id: string) => Promise<void>,
+  }
+
+  constructor(app: any) {
+    this._app = app
+    app.on('start', () => {
+      app.registerJobExecutor(
+        METADATA_BACKUP_JOB_TYPE,
+        this._executor.bind(this)
+      )
+    })
+  }
+
+  async _executor({ cancelToken, job: job_, schedule }): Executor {
+    if (schedule === undefined) {
+      throw new Error('backup job cannot run without a schedule')
+    }
+
+    const job: MetadataBackupJob = (job_: any)
+    const remoteIds = unboxIdsFromPattern(job.remotes)
+    if (remoteIds.length === 0) {
+      throw new Error('metadata backup job cannot run without remotes')
+    }
+
+    const poolIds = unboxIdsFromPattern(job.pools)
+    const isEmptyPools = poolIds.length === 0
+    if (!job.xoMetadata && isEmptyPools) {
+      throw new Error('no metadata mode found')
+    }
+
+    const app = this._app
+    const { retentionXoMetadata, retentionPoolMetadata } =
+      job?.settings[schedule.id] || {}
+
+    const timestamp = Date.now()
+    const formattedTimestamp = safeDateFormat(timestamp)
+    const commonMetadata = {
+      jobId: job.id,
+      jobName: job.name,
+      scheduleId: schedule.id,
+      scheduleName: schedule.name,
+      timestamp,
+    }
+
+    const files = []
+    if (job.xoMetadata && retentionXoMetadata > 0) {
+      const xoMetadataDir = `xo-config-backups/${schedule.id}`
+      const dir = `${xoMetadataDir}/${formattedTimestamp}`
+
+      const data = JSON.stringify(await app.exportConfig(), null, 2)
+      const fileName = `${dir}/data.json`
+
+      const metadata = JSON.stringify(commonMetadata, null, 2)
+      const metaDataFileName = `${dir}/metadata.json`
+
+      files.push({
+        executeBackup: defer(($defer, handler) => {
+          $defer.onFailure(() => handler.rmtree(dir))
+          return Promise.all([
+            handler.outputFile(fileName, data),
+            handler.outputFile(metaDataFileName, metadata),
+          ])
+        }),
+        dir: xoMetadataDir,
+        retention: retentionXoMetadata,
+      })
+    }
+    if (!isEmptyPools && retentionPoolMetadata > 0) {
+      files.push(
+        ...(await Promise.all(
+          poolIds.map(async id => {
+            const poolMetadataDir = `xo-pool-metadata-backups/${
+              schedule.id
+            }/${id}`
+            const dir = `${poolMetadataDir}/${formattedTimestamp}`
+
+            // TODO: export the metadata only once then split the stream between remotes
+            const stream = await app.getXapi(id).exportPoolMetadata(cancelToken)
+            const fileName = `${dir}/data`
+
+            const pool = this._app.getObject(id)
+            const metadata = JSON.stringify(
+              {
+                ...commonMetadata,
+                poolDescription: pool.name_description,
+                poolId: id,
+                poolName: pool.name_label,
```

Comment:
A standalone pool does not have a name.

Maybe we should simply had the pool and pool master record.